packages/@aws-cdk/aws-elasticloadbalancingv2/lib/shared/imported.ts

```diff
@@ -29,6 +33,7 @@ export abstract class ImportedTargetGroupBase extends CoreConstruct implements I
     super(scope, id);
 
     this.targetGroupArn = props.targetGroupArn;
+    this.targetGroupName = targetGroupNameFromArn(props.targetGroupArn) || cdk.Aws.NO_VALUE;
```

Comment:
I don't think you need the `||` part (`targetGroupNameFromArn` never returns `undefined`).